src/main/java/com/minecolonies/coremod/commands/ActionArgumentType.java

```diff
@@ -99,9 +128,34 @@ public int allowedSpaceCount()
         return citizenNameList;
     }
 
+    @NotNull
+    private static List<String> getCitizenIds(@Nullable final Colony colonyToUse)
+    {
+        final List<Colony> colonyList;
+        if (null != colonyToUse)
+        {
+            colonyList = Collections.singletonList(colonyToUse);
+        }
+        else
+        {
+            colonyList = ColonyManager.getColonies();
+        }
+        final List<String> citizenNameList = new ArrayList<>();
+        for (final Colony colony : colonyList)
+        {
+            final List<CitizenData> citizenDataList = colony.getCitizenManager().getCitizens();
+            for (final CitizenData citizenData : citizenDataList)
+            {
+                citizenNameList.add(String.valueOf(citizenData.getId()));
+            }
+        }
+        return citizenNameList;
+    }
+
+    @NotNull
     public List<String> getTabCompletions(@NotNull final MinecraftServer server,
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of returns of this method 7, down to the maximum allowed 6. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1142)